src/libraries/pkg/test/project.csproj.template

```diff
@@ -7,13 +7,27 @@
     <TestPackageVersion>{PackageVersion}</TestPackageVersion>
     <!-- Turn off end of life target framework checks as we intentionally build older .NETCoreApp configurations. -->
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
+    <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
 
     <SupportFilesDir Condition="'$(SupportFilesDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Directory.Build.props))\support</SupportFilesDir>
   </PropertyGroup>
-  
+
   <ItemGroup>
     <PackageReference Include="{PackageId}" Version="{PackageVersion}" />
   </ItemGroup>
 
+  <ItemGroup>
+    <KnownFrameworkReference Include="Microsoft.NETCore.App"
+      DefaultRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      IsTrimmable="true"
+      LatestRuntimeFrameworkVersion="5.0.0-rc.1.20451.14"
+      RuntimeFrameworkName="Microsoft.NETCore.App"
+      RuntimePackNamePatterns="Microsoft.NETCore.App.Runtime.**RID**"
+      RuntimePackRuntimeIdentifiers="linux-arm;linux-arm64;linux-musl-arm64;linux-musl-x64;linux-x64;osx-x64;rhel.6-x64;tizen.4.0.0-armel;tizen.5.0.0-armel;win-arm;win-arm64;win-x64;win-x86;ios-arm64;ios-arm;ios-x64;ios-x86;tvos-arm64;tvos-x64;android-arm64;android-arm;android-x64;android-x86;browser-wasm"
+      TargetFramework="net6.0"
+      TargetingPackName="Microsoft.NETCore.App.Ref"
+      TargetingPackVersion="5.0.0-rc.1.20451.14"/>
```

Comment:
Don't hardcode these versions and once you move this to `frameworkSettings/net6.0` use `RuntimeFrameworkVersion`.